### DISCUSSION TITLE
MPI Intel for PETCs

### DISCUSSION THREAD

**Discussion by jinca** [idaholab/moose#16425] [Q&A Getting Started] [answered]
Hello, 
Have you tried to run MPI Intel with MKL with PETSc? Specially the version 2020.2?
I got this error:
"[0] MPI startup(): I_MPI_OFA_TRANSLATION_CACHE variable has been removed from the product, its value is ignored"
Thanks in advance,
Julita Inca

**Comment by milljm** [comment]
Because I have a little background into your other error(s)...

I think the TestHarness error you mentioned to me, is due to Python 3.8 (possibly). Can you produce the error from one of the 50 that failed?

Thanks!

**Reply by milljm** [reply]
We might need to take a look at the libMesh build/configure log. Though I am no expert like @lindsayad. Maybe he doesn't need it...

Still, would you be able to provide that log? It should be located at:

```pre
/home/yulys/projects/moose/libmesh/build/config.log
```

Thanks!

**Comment by fdkong** [comment, marked as answer]
Just trying to answer your original question. Yes, PETSc in general work very well with MKL. But in MOOSE, by default, we do not choose to use MKL. You need to specify blaslpack dir to pick up MKL (`--with-blas-lib=where/your/MKL` and   `--with-lapack-lib=where/your/MKL`).

**Comment by jinca** [comment]
Thank you for the answers. Apparently is a permission problem 
*******************************************************************************
         UNABLE to CONFIGURE with GIVEN OPTIONS    (see configure.log for details):
-------------------------------------------------------------------------------
Error running make on PTScotch: Could not execute "['cd /home/ir-inca1/projects/moose/petsc/arch-moose/externalpackages/git.ptscotch/src && make clean ptesmumps esmumps']":
/usr/bin/mkdir -p ../bin
/usr/bin/mkdir -p ../include
/usr/bin/mkdir -p ../lib
(cd libscotch ;      make clean)
make[1]: Entering directory `/home/ir-inca1/projects/moose/petsc/arch-moose/externalpackages/git.ptscotch/src/libscotch'
rm -f *~ *.o lib*.a parser_yy.c parser_ly.h parser_ll.c *scotch.h *scotchf.h y.output *dummysizes
make[1]: Leaving directory `/home/ir-inca1/projects/moose/petsc/arch-moose/externalpackages/git.ptscotch/src/libscotch'
(cd scotch ;         make clean)
make[1]: Entering directory `/home/ir-inca1/projects/moose/petsc/arch-moose/externalpackages/git.ptscotch/src/scotch'
rm -f *~ *.o acpl amk_ccc amk_fft2 amk_grf amk_hy amk_m2 amk_p2 atst gbase gcv *ggath *gmap gmk_hy gmk_m2 gmk_m3 gmk_msh gmk_ub2 gmtst *gord gotst gout *gpart *gscat *gtst mcv mmk_m2 mmk_m3 mord mtst
make[1]: Leaving directory `/home/ir-inca1/projects/moose/petsc/arch-moose/externalpackages/git.ptscotch/src/scotch'
(cd libscotchmetis ; make clean)
make[1]: Entering directory `/home/ir-inca1/projects/moose/petsc/arch-moose/externalpackages/git.ptscotch/src/libscotchmetis'
rm -f *~ *.o lib*.a metis.h parmetis.h
make[1]: Leaving directory `/home/ir-inca1/projects/moose/petsc/arch-moose/externalpackages/git.ptscotch/src/libscotchmetis'
(cd check ;          make clean)
make[1]: Entering directory `/home/ir-inca1/projects/moose/petsc/arch-moose/externalpackages/git.ptscotch/src/check'
rm -f *~ *.o
make[1]: Leaving directory `/home/ir-inca1/projects/moose/petsc/arch-moose/externalpackages/git.ptscotch/src/check'
(cd esmumps ;        make clean)
make[1]: Entering directory `/home/ir-inca1/projects/moose/petsc/arch-moose/externalpackages/git.ptscotch/src/esmumps'
rm -f *~ common.h *.o lib*.a main_esmumps
make[1]: Leaving directory `/home/ir-inca1/projects/moose/petsc/arch-moose/externalpackages/git.ptscotch/src/esmumps'
(cd libscotch ;      make VERSION=6 RELEASE=0 PATCHLEVEL=9 scotch && make install)

**Reply by fdkong** [reply]
Could you share "configure.log" with us?

**Reply by jinca** [reply]
Thanks Fande, give me some minutes, I was trying to install it with GCC7 instead of the Intel packages

**Reply by fdkong** [reply]
Yes, if you are using intel compilers, you will be fine with `petsc` and `libmesh`, but you might have troubles with `moose`.

`MOOSE` does not support intel compilers anymore. It is better to use `GCC`
